sr_hand_health_report/src/sr_hand_health_report/backlash_check.py

```diff
@@ -0,0 +1,214 @@
+
+            for joint in finger_object.joints_dict.values():
+                rospy.logwarn(f"Running check for {joint.joint_name}")
+                if finger_object.finger_name in ('ff', 'mf', 'rf', 'lf') and joint.joint_index != "j1":
+                    joint_result = self.wiggle_joint(joint)
+                    result['backlash'][joint.joint_name] = joint_result
+                elif finger_object.finger_name in ('th', 'wr'):
+                    joint_result = self.wiggle_joint(joint)
+                    result['backlash'][joint.joint_name] = joint_result
+
+                if self._stopped_execution:
+                    break
+            if self._stopped_execution:
+                return
+
+            self.switch_controller_mode("position")
+            self.move_finger_to_side(finger_object, 'left')
+
+        self._result = result
+        self._stopped_execution = True
+        return
+
+    def move_finger_to_side(self, finger_object, side):
+        """
+            Moves the finger to the left/right joint limit of J4
+            @param finger_object: Finger object indicating which finger to move
+            @param side: String defining the side, 'right' or 'left'
+        """
+        angle = math.radians(-20) if side == 'right' else math.radians(20)
+        angle *= self._side_sign_map[finger_object.finger_name]
+        if "J4" in finger_object.joints_dict:
+            finger_object.joints_dict['J4'].move_joint(angle, 'position')
+
+    def wiggle_joint(self, joint):
+        """
+            Moves the joint repeatably by switching the PWM sign and sending the commands to the effort controller.
+            The result is being save into the _result variable and contains the average and standard deviation of
+            the collected over time durations between move direction changes.
+            @param joint: Joint object indicating which joint to 'wiggle'
+        """
+        test_times = []
+        total_time = None
+
+        test_start_time = rospy.get_rostime()
+        timeout_condition = True
+
+        while len(test_times) < self.TEST_COUNT and timeout_condition and not rospy.is_shutdown():
+
+            success = True
+            test_time = rospy.get_rostime()
+            difference = 0
+            starting_position = joint.get_current_position()
+            start_time = rospy.get_rostime()
+            joint.move_joint(-self.FINGER_PWM, 'effort')
+            current_time = 0
+            while abs(difference) < self.THRESHOLD and not rospy.is_shutdown():
+                difference = joint.get_current_position() - starting_position
+                current_time = rospy.get_rostime() - start_time
+                time.sleep(0.01)
+                if current_time > rospy.Duration.from_sec(self.WIGGLING_TIME):
+                    test_time += rospy.Duration.from_sec(self.WIGGLING_TIME)
+                    success = False
+                    break
+
```

Comment:
I don't understand why this is done twice. I am wondering if it would make sense to encapsulate it in a function because the code is repeated twice.